include/glowl/Mesh.hpp

```diff
@@ -48,35 +48,80 @@ namespace glowl
     public:
         typedef std::unique_ptr<BufferObject> BufferObjectPtr;
 
+        template<typename VertexDataT>
+        using VertexInfo = std::pair<std::vector<VertexDataT>, glowl::VertexLayout>;
+
+        template<typename VertexDataT>
+        using VertexPtrInfo = std::tuple<VertexDataT*, std::size_t, VertexLayout>;
+
+        template<typename VertexDataT>
+        using VertexInfoList = std::vector<VertexInfo<VertexDataT>>;
+
+        template<typename VertexDataT>
+        using VertexPtrInfoList = std::vector<VertexPtrInfo<VertexDataT>>;
+
         /**
          * \brief Mesh constructor that requires data pointers and byte sizes as input.
          *
          * Note: Active OpenGL context required for construction.
          * Use std::unqiue_ptr (or shared_ptr) for delayed construction of class member variables of this type.
          */
-        template<typename VertexPtr, typename IndexPtr>
-        Mesh(std::vector<VertexPtr> const&    vertex_data,
-             std::vector<size_t> const&       vertex_data_byte_sizes,
-             IndexPtr const                   index_data,
-             GLsizeiptr const                 index_data_byte_size,
+        template<typename VertexDataT, typename IndexDataT>
+        Mesh(std::vector<VertexDataT*> const& vertex_data,
+             std::vector<std::size_t> const&  vertex_data_byte_sizes,
              std::vector<VertexLayout> const& vertex_descriptor,
+             IndexDataT* const                index_data,
+             std::size_t const                index_data_byte_size,
              GLenum const                     index_type = GL_UNSIGNED_INT,
-             GLenum const                     usage = GL_STATIC_DRAW,
-             GLenum const                     primitive_type = GL_TRIANGLES);
+             GLenum const                     primitive_type = GL_TRIANGLES,
+             GLenum const                     usage = GL_STATIC_DRAW);
+
+        /**
+         * \brief Mesh constructor that requires data pointers and byte sizes as input.
+         *
+         * \param vertex_info A vector where each entry contains (in a tuple) a pointer to vertex data,
+         *  the byte size of that data and the corresponding VertexLayout
+         *
+         * Note: Active OpenGL context required for construction.
+         * Use std::unqiue_ptr (or shared_ptr) for delayed construction of class member variables of this type.
+         */
+        template<typename VertexDataT, typename IndexDataT>
+        Mesh(VertexPtrInfoList<VertexDataT> const& vertex_info,
+             IndexDataT* const                     index_data,
```

Comment:
I think this should be `IndexDataT const*` to match the call to BufferObject constructor and eventually glNamedBufferData.